### ISSUE TITLE
你好，whl文件安装后不可用，test.py不通过

### ISSUE THREAD

**Issue by smile-care** [OpenGVLab/InternImage#107]

**Comment by czczup** [comment]
你好，能贴一下具体报错的log吗

**Comment by xiaoyaod** [comment]
pytorch                   1.12.1          py3.8_cuda11.3_cudnn8.3.2_0    pytorch  这边我用的conda创建的
DCNv3-1.0+cu113torch1.12.1-cp38-cp38-linux_x86_64.whl 
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device

**Comment by xiaoyaod** [comment]
out = deform_conv2d(input, offset, weight, mask=mask,padding=1)
weight 变成 shape（out,1,k_h,k_w）全为1 的固定参数 mask softmax沿着kernel_h*kernel_w做softmax 是否和您论文 代码中dcnv3_core_pytorch 的一样

**Comment by dangerous-xu** [comment]
遇到相同的报错
环境描述
[*] Ubuntu16.04
[*] RTX2080Ti
[*] CUDA 11.3
[*] torch==1.11.0+cu113 torchvision==0.12.0+cu113
[*] DCNv3 mask.sh报错，后改用whl安装
[*] 运行./ops_dcbv3/test.py报错

‘’‘
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
>>> forward double
* False check_forward_equal_with_pytorch_double: max_abs_err 2.87e-03 max_rel_err nan
>>> forward float
* False check_forward_equal_with_pytorch_float: max_abs_err 3.72e-03 max_rel_err nan
>>> backward double: channels 1
* False input_grad check_backward_equal_with_pytorch_double: max_abs_err 2.82e-01 max_rel_err nan
* False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 3.97e-03 max_rel_err nan
* False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 7.91e-03 max_rel_err nan
>>> backward double: channels 16
* False input_grad check_backward_equal_with_pytorch_double: max_abs_err 4.44e-01 max_rel_err nan
* False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 4.24e-02 max_rel_err nan
* False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 9.69e-02 max_rel_err nan
>>> backward double: channels 30
* False input_grad check_backward_equal_with_pytorch_double: max_abs_err 3.18e-01 max_rel_err nan
* False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 8.77e-02 max_rel_err nan
* False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 1.73e-01 max_rel_err nan
‘’‘

**Comment by JiaweiZhao-git** [comment]
+1, also meet this error

**Comment by jeonga0303** [comment]
Does anyone know the answer?

**Comment by SSZ1** [comment]
waiting for solutions......

**Comment by smile-care** [comment]
> 遇到相同的报错 环境描述 [_] Ubuntu16.04 [_] RTX2080Ti [_] CUDA 11.3 [_] torch==1.11.0+cu113 torchvision==0.12.0+cu113 [_] DCNv3 mask.sh报错，后改用whl安装 [_] 运行./ops_dcnv3/test.py报错
> 
> ‘’‘ error in dcnv3_im2col_cuda: no kernel image is available for execution on the device error in dcnv3_im2col_cuda: no kernel image is available for execution on the device error in dcnv3_im2col_cuda: no kernel image is available for execution on the device error in dcnv3_im2col_cuda: no kernel image is available for execution on the device error in dcnv3_im2col_cuda: no kernel image is available for execution on the device
> 
> > > > forward double
> 
> * False check_forward_equal_with_pytorch_double: max_abs_err 2.87e-03 max_rel_err nan
> 
> > > > forward float
> 
> * False check_forward_equal_with_pytorch_float: max_abs_err 3.72e-03 max_rel_err nan
> 
> > > > backward double: channels 1
> 
> * False input_grad check_backward_equal_with_pytorch_double: max_abs_err 2.82e-01 max_rel_err nan
> * False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 3.97e-03 max_rel_err nan
> * False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 7.91e-03 max_rel_err nan
> 
> > > > backward double: channels 16
> 
> * False input_grad check_backward_equal_with_pytorch_double: max_abs_err 4.44e-01 max_rel_err nan
> * False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 4.24e-02 max_rel_err nan
> * False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 9.69e-02 max_rel_err nan
> 
> > > > backward double: channels 30
> 
> * False input_grad check_backward_equal_with_pytorch_double: max_abs_err 3.18e-01 max_rel_err nan
> * False offset_grad check_backward_equal_with_pytorch_double: max_abs_err 8.77e-02 max_rel_err nan
> * False mask_grad check_backward_equal_with_pytorch_double: max_abs_err 1.73e-01 max_rel_err nan
>   ‘’‘

是的，我一样的环境，不行
